### PR TITLE
Use period serialization extension

### PR DESCRIPTION
**Pull request by vegaro**
Added a `convertToMap` function that can be reused

### CODE REVIEW

**Comment by aboedo** [comment]
@vegaro what's missing from this one?

**Comment by vegaro** [comment]
Nothing, I totally forgot about it

File: android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt [hunk 1]
```diff
@@ -258,6 +254,15 @@ fun List<*>.convertToJsonArray(): JSONArray {
     return writableArray
 }
 
+fun JSONObject.convertToMap(): Map<String, String?> =
+    this.keys().asSequence<String>().associate { key ->
+        if (this.isNull(key)) {
```
Comment (aboedo) [inline android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt:259]: will this map null to null? would we want to skip instead?

Comment (vegaro) [reply android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt:259]: It maps `JSON.Null` to `null`, and we want the nulls in the map

**Comment by vegaro** [comment]
We can batch this one with the other PR for release

**Comment by aboedo** [comment]
might wanna do it separately, since the other one changes source files locations and requires a bit more work when integrating

**Comment by vegaro** [comment]
ok, I will do it now then

**Pull request merged**